cmd/local.go

```diff
@@ -199,8 +232,11 @@ func (l LocalModel) Cleanup(channels *turnstile.ChannelMap) {
 	//Gitignore operations
 	createNewGitIgnoreFile(l.Nonmem)
 
+	// this should have been either completed well before, or must at least wait now to complete the hash
+	// before writing out the config
+	l.Nonmem.DataMD5 = <-hashChan
```

Comment:
Wouldn't it be better to have wait group here? Or at least a select with a maximum read count? Just to make sure that we are certain the operation has completed? Here we run the risk of pulling nothing off the channel and continuing on

Reply:
The point of having that unbuffered channel is such that it will wait until something is on the channel to pull from it though...

> If the capacity is zero or absent, the channel is unbuffered and communication succeeds only when both a sender and receiver are ready.
> If the channel is unbuffered, the sender blocks until the receiver has received the value

```
package main

import (
	"fmt"
	"time"
)

func main() {
	exChan := make(chan string)
	startTime := time.Now()
	go func() {
		sleepTime := 2 * time.Second
		fmt.Printf("in go routine, about to sleep for %s seconds\n\n", sleepTime)
		time.Sleep(sleepTime)
		exChan <- "message on channel"
		fmt.Println("finished in goroutine")
	}()
	fmt.Println("printing before pulling from channel")
	fmt.Println(<-exChan)
	fmt.Println("printing after pulling from channel %s seconds later", time.Since(startTime))

}

```

```
➜ go run channel-wait.go
printing before pulling from channel
in go routine, about to sleep for 2s seconds

finished in goroutine
message on channel
printing after pulling from channel %s seconds later 2.004226966s
```

Reply:
You're right. Didn't pay enough attention to see that it was unbuffered. It's a personal paranoia point of mine to have cancellation contexts and wait groups. I have to deal with this in tests all the time too. Should have paid more attention.